src/WebJobs.Script.WebHost/Security/Authorization/Policies/AuthorizationOptionsExtensions.cs

```diff
@@ -49,6 +51,33 @@ public static void AddScriptPolicies(this AuthorizationOptions options)
                     return false;
                 });
             });
+
+            options.AddPolicy(PolicyNames.SystemKeyAuthLevel, p =>
+            {
+                p.AddScriptAuthenticationSchemes();
+                p.RequireAssertion(c =>
+                {
+                    if (c.Resource is AuthorizationFilterContext filterContext)
+                    {
+                        if (filterContext.HttpContext.Request.IsAppServiceInternalRequest())
+                        {
+                            return true;
+                        }
+
+                        object extensionName = filterContext.RouteData.Values["extensionName"];
+                        if (extensionName != null)
```

Comment:
Just so this is not tied to extensions, can we also look for a `keyname` value in route data for other scenarios (so if extension name is not present, try key name directly)

Reply:
`keyName=$"{extensionName}_extension"`

Did you mean authorizing by `code`? We already have it here:
https://github.com/Azure/azure-functions-host/blob/40890abf10b2d7e41ff6b1c84baf85cf0c3e9949/src/WebJobs.Script.WebHost/Security/Authorization/Policies/AuthorizationOptionsExtensions.cs#L19

Reply:
System should always be based on named keys. We can't remove that now as it would be a breaking change, but this policy needs to be able to handle that as well.